### DISCUSSION TITLE
Run `dependsOn` tasks of a remote's `serve` target when it's specified in `devRemotes`

### DISCUSSION THREAD

**Discussion by royling** [nrwl/nx#23318] [Feature Requests]
Within a module federation workspace, we usually start the local dev server by running `nx serve host --devRemotes=remote`.
If the `serve` target of the remote has a dependent task specified in `dependsOn` like below:
```json
"serve": {
  "executor": "@nx/angular:dev-server",
  "dependsOn": ["custom"],
  ...
},
"custom": {
  "executor": "nx:run-commands",
  ...
}
```
The `custom` target is not executed for the `remote:serve`, this makes `--devRemotes` option less helpful.
> When running `nx serve remote`, it takes `dependsOn` into the task graph and runs it as expected.

I understand that this may increase the complexity for handling the task graph, especially it may result to a large depth of dependency levels (custom targets may have their `dependsOn`).

So, I opened this discussion to see whether this is a reasonable feature request. What's your opinion on it?

**Comment by royling** [comment]
Although I found this in a module federation workspace, I think it also applies to a standalone project with `build` and `serve` targets.
As you know, the `serve` target has the `buildTarget` option to specify the target dependency within the same project. If both depends on a `custom` target, now we have to add `dependsOn` to both in order that the `custom` target is executed in both `nx build` and `nx serve`:
```json
"build": {
  "executor": "@nx/angular:webpack-browser",
  "dependsOn": ["custom"],
  ...
},
"serve": {
  "executor": "@nx/angular:module-federation-dev-server",
  "dependsOn": ["custom"],
  ...
  "configurations": {
    "production": {
      "buildTarget": "shell:build:production"
    }
  }
}
```
This seems counter-intuitive given that the `buildTarget` is specified for `serve` and the `build` target has its `dependsOn`, shouldn't the `build` target handle its dependent tasks?

**Comment by jaysoo** [comment]
@royling Thanks for the report. Since we're using the `runExecutor` function to run dev remotes, it is skipping the dependencies as you've discovered.

The workaround could be the wrap both commands into another target, such as:


```
"serve-with-dep": {
  "command": "nx run <app>:custom && nx run <app>:serve"
}
```

I haven't tested this myself, but I image it should work.

**Reply by royling** [reply]
@jaysoo Thanks for the reply and workaround, but the wrapper target cannot be integrated with `--devRemotes` option.
Currently my workaround is to run the `custom` target manually before `nx serve host --devRemotes remote`, it worked for me.

If that's by design, I'll close this discussion.
